muse_payment_adaptor/api_interface/muse_api_client.py

```diff
@@ -0,0 +1,158 @@
+        """
+        raise NotImplementedError("`api_endpoint` not implemented")
+
+    def _submit(self, model: BaseModel) -> str:
+        response = self.api_http_client.send_post_request(self._api_endpoint, model.json())
+        if response.ok:
+            return response.content.decode('utf-8')
+        else:
+            logger.error(f"Muse request failed, HTTP code {response.status_code}")
+
+    def _build_message_signature(self, message: BaseModel) -> str:
+        # To JSON, To UTF-8 bytes
+        encoded_msg = json.dumps({'message': message.dict()}).encode('utf-8')
+        # Create signature, To b64 bytes, To b64 string (ascii)
+        signature_str = create_signature_b64_string(encoded_msg)
+        return signature_str
+
+
+# endregion
+# region BP
+
+class BulkPaymentSubmissionClient(MuseApiClient):
+    _api_endpoint = MusePaymentAdaptorConfig.muse_post_bp_endpoint
+
+    def submit_bulky_payment(self, bill: Bill, payment_desc: str = '') \
+            -> Union[Tuple[PaymentRequest, str], Tuple[None, None]]:
+        """
+        Submit Bill as bulky payment to MUSE Api. This method will send the submission and save the request in the
+        database.
+
+        :param bill: Bill to be submitted
+        :param payment_desc: Payment description to be included with the submission
+        :return: tuple of PaymentRequest database model and response content, if the request is successful
+        """
+        try:
+            request = self._build_request(bill, payment_desc)
+            request_model = self._store_request(request)
+            # TODO update model after submit?
+            return request_model, self._submit(request)
+        except BaseException as e:
+            logger.error("Error while sending MUSE request", exc_info=e)
+            return None, None
+
+    def _build_request_message(self, bill: Bill, description: str) -> BulkyPaymentSubmissionRequestMessage:
+        message_header = self._build_request_message_header(bill)
+        message_content = self._build_request_message_summary(bill, description)
+        pay_list = self._build_request_pay_list(bill)
+        return BulkyPaymentSubmissionRequestMessage(
+            messageHeader=message_header,
+            paymentSummary=message_content,
+            payList=pay_list,
+        )
+
+    def _build_request(self, bill: Bill, description: str) -> BulkyPaymentSubmissionRequest:
+        message = self._build_request_message(bill, description)
+        signature = self._build_message_signature(message)
+        return BulkyPaymentSubmissionRequest(
+            message=message,
+            digitalSignature=signature
+        )
+
+    def _build_request_message_header(self, bill: Bill) -> BulkyPaymentMessageHeader:
+        return BulkyPaymentMessageHeader(
+            sender=MusePaymentAdaptorConfig.muse_api_sender,
+            receiver=MusePaymentAdaptorConfig.muse_api_receiver,
+            msgId=str(bill.uuid),
```

Comment:
I remember we have discrepancy between our uuids and mssql unique identifier. I think this one will be char(32) format. Will it work without any adjustments?

Reply:
Yes